engine/apps/grafana_plugin/views/status.py

```diff
@@ -1,19 +1,81 @@
 from django.apps import apps
 from django.conf import settings
+from django.http import JsonResponse
 from rest_framework.request import Request
 from rest_framework.response import Response
 from rest_framework.views import APIView
 
+from apps.auth_token.auth import PluginAuthentication
 from apps.grafana_plugin.helpers import GrafanaAPIClient
 from apps.grafana_plugin.permissions import PluginTokenVerified
+from apps.grafana_plugin.tasks.sync import plugin_sync_organization_async
 from apps.user_management.models import Organization
 from common.api_helpers.mixins import GrafanaHeadersMixin
 
 
 class StatusView(GrafanaHeadersMixin, APIView):
     permission_classes = (PluginTokenVerified,)
 
+    def post(self, request: Request) -> Response:
+        """
+        Called asyncronounsly on each start of the plugin
+        Checks if plugin is correctly installed and async runs a task
+        to sync users, teams and org
+        """
+        # Check if the plugin is currently undergoing maintenance, and return response without querying db
+        if settings.CURRENTLY_UNDERGOING_MAINTENANCE_MESSAGE:
+            return JsonResponse(
+                {
+                    "currently_undergoing_maintenance_message": settings.CURRENTLY_UNDERGOING_MAINTENANCE_MESSAGE,
+                }
+            )
+
+        stack_id = self.instance_context["stack_id"]
+        org_id = self.instance_context["org_id"]
+
+        is_installed = False
+        token_ok = False
+        allow_signup = True
+
+        # Check if organization is in OnCall database
+        if organization := Organization.objects.get(stack_id=stack_id, org_id=org_id):
+            is_installed = True
+            # _, resp = GrafanaAPIClient(api_url=organization.grafana_url, api_token=organization.api_token).check_token()
+            # token_ok = resp["connected"]
+            # TODO: Consider checking from OnCall db instead of Grafana to make it faster:
+            token_ok = organization.api_token_status == Organization.API_TOKEN_STATUS_OK
+        else:
+            DynamicSetting = apps.get_model("base", "DynamicSetting")
+            allow_signup = DynamicSetting.objects.get_or_create(
+                name="allow_plugin_organization_signup", defaults={"boolean_value": True}
+            )[0].boolean_value
+
+        # Check if current user is in OnCall database
+        user_is_present_in_org = PluginAuthentication.is_user_from_request_present_in_organization(
+            request, organization
+        )
+        # If user is not present in OnCall database, set token_ok to False, which will trigger reinstall
+        if not user_is_present_in_org:
+            token_ok = False
+
+        # Start task to refresh organization data in OnCall database with Grafana
+        plugin_sync_organization_async.apply_async((organization.pk,))
```

Comment:
The task itself does not set `api_token_status` so we should set it to PENDING here so that if multiple users are loading the plugin at the same time we don't build up unnecessary tasks.

Reply:
add same as it was in POST /sync